File: src/main/java/net/unaussprechlich/managedgui/lib/databases/Player/PlayerModel.java
/*
 * ***************************************************************************
 *
 *         Copyright © 2016 unaussprechlich - ALL RIGHTS RESERVED
 *
 * ***************************************************************************
 */

package net.unaussprechlich.managedgui.lib.databases.Player;

import net.minecraft.util.ResourceLocation;
import net.unaussprechlich.managedgui.lib.databases.Player.data.Rank;

import java.util.UUID;

/**
 * PlayerModel Created by Alexander on 27.02.2017.
 * Description:
 **/
public class PlayerModel {

    private final String name;
    private final Rank rank;
    private final UUID uuid;
    private final ResourceLocation headResource;

    public PlayerModel(String name, UUID uuid, Rank rank, ResourceLocation headResource) {
        this.name = name;
        this.uuid = uuid;
        this.rank = rank;
        this.headResource = headResource;
    }

    public String getRankName(){
        if(rank.getRankFormatted().equalsIgnoreCase(rank.getRankColor().toString())) return rank.getRankColor() + name;
        return rank.getRankFormatted() + " " +  name;
    }

    public String getName() {
        return name;
    }

    public UUID getUuid() {
        return uuid;
    }

    public ResourceLocation getHeadResource() {
        return headResource;
    }
}

File: src/main/java/net/unaussprechlich/managedgui/lib/databases/player/PlayerModel.kt
/*
 * ***************************************************************************
 *
 *         Copyright © 2016 unaussprechlich - ALL RIGHTS RESERVED
 *
 * ***************************************************************************
 */

package net.unaussprechlich.managedgui.lib.databases.player

import net.minecraft.util.ResourceLocation
import net.unaussprechlich.managedgui.lib.databases.player.data.Rank
import net.unaussprechlich.managedgui.lib.handler.ResourceHandler
import net.unaussprechlich.managedgui.lib.util.resolver.NameResolver
import net.unaussprechlich.managedgui.lib.util.resolver.PlayerHeadResolver
import java.util.*

/**
 * PlayerModel Created by Alexander on 27.02.2017.
 * Description:
 */
class PlayerModel(val nameHistory : NameResolver.NameHistory, val uuid: UUID) {

    val name : String = nameHistory.currentName

    var rank : Rank? = null

    private var playerHead : ResourceLocation? = null


    var rankName: String = ""
        get() {
            if (rank!!.rankFormatted.equals(rank!!.rankColor.toString(), ignoreCase = true)) return rank!!.rankColor.toString() + name
            return rank!!.rankFormatted + " " + nameHistory.currentName
        }

    fun getPlayerHeadLoc(callback: (ResourceLocation) -> Unit){
        if(playerHead != null)
            callback.invoke(playerHead!!)
        else
            PlayerHeadResolver(name){ res ->
                playerHead = res ?: ResourceHandler.STEVE_HEAD.res
                callback.invoke(playerHead!!)
            }
    }
}
